src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,12 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    @Override
+    public List<? extends Machine> get() {
```

Comment:
we should be using a specified class instead of upper bounding. e.g. <Truck> for TruckProducer and etc.